File: Frontend/src/services/QuizService.js
import axios from 'axios';
import {base_url} from './base_url';
import { JwtToken } from './JwtToken';

class QuizService{

    addQuiz(quiz){
        return axios.post(`${base_url}/quiz-service/quiz/`,quiz,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

    GetAllQuiz(){
        return axios.get(`${base_url}/quiz-service/quiz/`,
        {
            headers:{
                jwtToken:JwtToken
            }
        });
    }

}

export default new QuizService();

File: Backend/quiz/src/main/java/com/quiz/service/QuizService.java
package com.quiz.service;

import java.util.*;

import com.quiz.model.Quiz;

public interface QuizService {
	
	// Add Quiz
	public Quiz addQuiz(Quiz quiz);
	
	// Update Quiz
	public Quiz updateQuiz(Quiz quiz);
	
	// Get All Quizzes
	public Set<Quiz> getQuizzes();
	
	// Get Single Quiz
	public Quiz getSingleQuiz(Long quizId);
	
	// Delete Quiz By Id
	public void deleteQuiz(Long quizId);

}
